C++: Add hello world documentation to language level README (#5209)
awsdocs/aws-doc-sdk-examples: cpp/README.rst

Building and running the code examples
=============
 

> The example code is organized in sub-folders by AWS service.
> For example "example_code/s3" contains the Amazon Simple Storage Service (Amazon S3) code examples.

Reviewer: ```suggestion
The example code is organized in subfolders by AWS service.
For example, "example_code/s3" contains the Amazon Simple Storage Service (Amazon S3) code examples.

```

Examples that use multiple services are located in the [example_code/cross-service](example_code/cross-service) folder.

> The examples use the CMake build system. Every service sub-folder contains a CMakeLists.txt file which will build all the examples for that service. 

Reviewer: ```suggestion
The examples use the CMake build system. Every service subfolder contains a CMakeLists.txt file that builds all the examples for that service. 

```

> Additional information geared specifically to understanding the example applications is at

Reviewer: ```suggestion
To understand more about the example applications, see

```

[Get started with the AWS SDK for C++ code examples](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started-code-examples.html).

Hello Service
-----------

Many of the services have a "Hello Service" folder with a starter project and README documentation. 

> For example, the AmazonS S3 Hello project is located at [example_code/s3/hello_s3](example_code/s3/hello_s3).

Reviewer: ```suggestion
For example, the Amazon S3 Hello project is located at [example_code/s3/hello_s3](example_code/s3/hello_s3).

```
